```CNDIT-1021``: Patient data post processing in the spring boot service

alog-enquizit opened the pull request:
- Add the post proessing logic for patient data.
- All the nested json data returned from the patient stored procedure is processed into the respective fields in the `PatientOP` object
- Added unit tests to validate all the processing segments

upasanapattnaik-eq commented on etl-data-pipeline/src/main/java/gov/cdc/etldatapipeline/changedata/model/dto/PatientOP.java:
```diff
@@ -46,96 +57,103 @@ public class PatientOP extends DebeziumMetadata {
+                                .orElse(null);
+                patientPhoneFn.apply("WP");
+                patientPhoneFn.apply("H");
+                patientPhoneFn.apply("C");
```
CP for cellphone

upasanapattnaik-eq reviewed (commented):
Looks good! This is a tricky piece of logic to move over. It is much appreciated.

alog-enquizit commented on etl-data-pipeline/src/main/java/gov/cdc/etldatapipeline/changedata/model/dto/patient/AddAuthUser.java:
```diff
@@ -0,0 +1,37 @@
+@NoArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AddAuthUser {
+    @JsonProperty("ADD_USER_ID")
```
Yeah that would be great!

alog-enquizit commented on etl-data-pipeline/src/main/java/gov/cdc/etldatapipeline/changedata/model/dto/PatientOP.java:
```diff
@@ -46,96 +57,103 @@ public class PatientOP extends DebeziumMetadata {
+
+            if (!ObjectUtils.isEmpty(patientEntityData)) {
+                Arrays.stream(mapper.readValue(patientEntityData, EntityData[].class))
+                        .filter(e -> e.getAssigningAuthorityCd().equalsIgnoreCase("SSA"))
```
Added the logic to handle other sections .

alog-enquizit commented on etl-data-pipeline/src/main/java/gov/cdc/etldatapipeline/changedata/model/dto/PatientOP.java:
```diff
@@ -1,18 +1,29 @@
-@ToString
-@EqualsAndHashCode
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class PatientOP extends DebeziumMetadata {
```
Yeah updated the consolidated object for both Patient and Provider

alog-enquizit commented on etl-data-pipeline/src/main/java/gov/cdc/etldatapipeline/changedata/model/dto/patient/EntityData.java:
```diff
@@ -0,0 +1,29 @@
+    private String assigningAuthorityCd;
+
+    public PatientOP updatePerson(PatientOP patient) {
+        patient.setPatientSsn(rootExtensionTxt);
```
Yeah using the max_entity_id

rseshadri99 commented:
@alog-enquizit Let us follow the "separation of concerns principle" for the Person Topic. Filter the Person Type and get the list of Id's (Patient,Provider,next of kin etc) and pass it to the corresponding stored procedure. This way if we have an issue with Patient dim only that sql procedure needs to be updated and limit scope of regression on other types of Person.

rseshadri99 commented:
Also the processPerson java method could be modularized and split as personEmail, personName, personAddress etc and call them if the object is not null

rseshadri99 commented on etl-data-pipeline/src/main/java/gov/cdc/etldatapipeline/changedata/model/dto/patient/AddAuthUser.java:
```diff
@@ -0,0 +1,37 @@
+    @JsonProperty("ADD_USER_ID")
+    private Long addUserId;
+    @JsonProperty("add_time")
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS", timezone = "UTC")
```
timezone = "UTC" @alog-enquizit Should this be UTC ? or should leave it to local

rseshadri99 replied:
@alog-enquizit Let us talk on this as a team - so we are clear if we should use timestamp with correct timezone based on JD's need

alog-enquizit commented:
> @alog-enquizit Let us follow the "separation of concerns principle" for the Person Topic. Filter the Person Type and get the list of Id's (Patient,Provider,next of kin etc) and pass it to the corresponding stored procedure. This way if we have an issue with Patient dim only that sql procedure needs to be updated and limit scope of regression on other types of Person.

Separated out the Patient and Provider logic. Addressed the feedback in the latest commit.

rseshadri99 commented:
> > Also the processPerson java method could be modularized and split as Email, Name, Address etc and call them if the object is not null @alog-enquizit
> 
> @rseshadri99 As mentioned this has been modularized

Looks good, thanks @alog-enquizit

alog-enquizit commented on data-reporting-service/src/main/java/gov/cdc/etldatapipeline/changedata/model/dto/persondetail/AddAuthUser.java:
```diff
@@ -0,0 +1,37 @@
+    @JsonProperty("ADD_USER_ID")
+    private Long addUserId;
+    @JsonProperty("add_time")
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS")
```
@rseshadri99 @upasanapattnaik-eq Deserializing the date time in the local timezone as discussed in slack.

rseshadri99 commented:
Looks good

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Anand Loganathan ***@***.***>
Sent: Wednesday, March 13, 2024 3:12:21 PM
To: CDCgov/NEDSS-DataAccess ***@***.***>
Cc: Ravindran Seshadri ***@***.***>; Mention ***@***.***>
Subject: Re: [CDCgov/NEDSS-DataAccess] CNDIT-1067: Patient data post processing in the spring boot service (PR #47)


Warning: untrusted sender domain

@alog-enquizit commented on this pull request.

________________________________

In data-reporting-service/src/main/java/gov/cdc/etldatapipeline/changedata/model/dto/persondetail/AddAuthUser.java<https://github.com/CDCgov/NEDSS-DataAccess/pull/47#discussion_r1523795370>:

> +import gov.cdc.etldatapipeline.changedata.model.dto.PersonExtendedProps;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDateTime;
+
***@***.***
***@***.***
***@***.***
***@***.***(ignoreUnknown = true)
+public class AddAuthUser {
+    @JsonProperty("ADD_USER_ID")
+    private Long addUserId;
+    @JsonProperty("add_time")
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS")


@rseshadri99<https://github.com/rseshadri99> @upasanapattnaik-eq<https://github.com/upasanapattnaik-eq> Deserializing the date time in the local timezone as discussed in slack.

—
Reply to this email directly, view it on GitHub<https://github.com/CDCgov/NEDSS-DataAccess/pull/47#pullrequestreview-1935017925>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A2AYNUOEBIPVKAD625OZSGTYYCQJLAVCNFSM6AAAAABENSTPNKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMZVGAYTOOJSGU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
